### PR TITLE
Organize vignettes, update site template

### LINKED ISSUE #704: [Feature Request]: Organise vignettes in sections
### Feature description

As the number of articles grows - it would be more much easier for a beginner users to navigate if they will be organised in some sections. A great example can be found here: https://github.com/cynkra/dm/tree/main/demo that renders into the following https://dplyr.tidyverse.org/articles/index.html

I don't have any ideas about the sections and their descriptions. Please decide.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by edelarua**
Closes #704 

This is a rough first attempt at grouping vignettes - please give suggestions for section names/layout/what should go where! @Melkiades @ayogasekaram

I have also updated the pkgdown site template to align with all the other NEST package sites.

### CODE REVIEW

**Comment by edelarua** [comment]
Preview of initial attempt:

rtables articles revamp

File: _pkgdown.yml [hunk 1]
```diff
@@ -1,37 +1,47 @@
...
     - clinical_trials
+
+  - title: Table Customization
+    navbar: Table Customization
+    contents:
+    - custom_appearance
+    - title_footer
+    - sorting_pruning
+    
+  - title: rtables Concepts
```
Comment (edelarua) [inline _pkgdown.yml:26]: Any ideas for this section name?

Comment (ayogasekaram) [reply _pkgdown.yml:26]: I think rtables concepts is good! It mentions the articles that are specific to rtables computations so it's correct and informative.

Comment (Melkiades) [reply _pkgdown.yml:26]: sounds good to me!

**Comment by edelarua** [comment]
Thanks for the suggestions! Here's iteration 2. 

Is there a better section to put the Exploratory Analysis vignette in? It's pretty basic information but not really necessary to know.

iter2

**Review by shajoezhu — APPROVED** [review]
lgtm! Thanks @edelarua for the re-org! this looks brilliant!

**Pull request merged**